### PR TITLE
Add codeowners per subproject

### PR DESCRIPTION
**Pull request by jirikuncar**
Prepare more automatisation.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,7 +2,24 @@
...
+/datadog/api/           @DataDog/integrations-tools-and-libraries
+/tests/integration/api/ @DataDog/integrations-tools-and-libraries
+/tests/unit/api/        @DataDog/integrations-tools-and-libraries
+
+# Dogshell
+/datadog/dogshell/      @DataDog/integrations-tools-and-libraries
+/tests/integration/dogshell/ @DataDog/integrations-tools-and-libraries
+
+# Dogstatd
+/datadog/dogstatsd/     @DataDog/agent-core
```
Comment (jirikuncar) [inline .github/CODEOWNERS]:
```suggestion
/datadog/dogstatsd/     @DataDog/integrations-tools-and-libraries @DataDog/agent-core
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -2,7 +2,24 @@
...
+/tests/unit/dogstatsd/  @DataDog/agent-core
```
Comment (jirikuncar) [inline .github/CODEOWNERS]:
```suggestion
/tests/unit/dogstatsd/  @DataDog/integrations-tools-and-libraries @DataDog/agent-core
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -2,7 +2,24 @@
...
+
+# Threadstats
+/datadog/threadstats/   @DataDog/agent-core
```
Comment (jirikuncar) [inline .github/CODEOWNERS]:
```suggestion
/datadog/threadstats/   @DataDog/integrations-tools-and-libraries @DataDog/agent-core
```

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -2,7 +2,24 @@
...
+/tests/unit/threadstats/ @DataDog/agent-core
```
Comment (jirikuncar) [inline .github/CODEOWNERS]:
```suggestion
/tests/unit/threadstats/ @DataDog/integrations-tools-and-libraries @DataDog/agent-core
```

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -2,7 +2,24 @@
 # Rules are matched bottom-to-top, so one team can own subdirectories
 # and another team can own the rest of the directory.
 
-# All your base
+# API
+/datadog/api/           @DataDog/integrations-tools-and-libraries
```
Comment (gzussa) [inline .github/CODEOWNERS]: Small detail but could we align everything here?

Comment (jirikuncar) [reply .github/CODEOWNERS]: done

**Pull request merged**